service/service.go

```diff
@@ -252,13 +255,38 @@ func New(config Config) (*Service, error) {
 	}
 
 	// These credentials will be used when creating AzureClients for Control Plane clusters.
-	gsClientCredentialsConfig, err := credential.NewAzureCredentials(
-		config.Viper.GetString(config.Flag.Service.Azure.ClientID),
-		config.Viper.GetString(config.Flag.Service.Azure.ClientSecret),
-		config.Viper.GetString(config.Flag.Service.Azure.TenantID),
-	)
-	if err != nil {
-		return nil, microerror.Mask(err)
+	var gsClientCredentialsConfig auth.ClientCredentialsConfig
+	{
+		mcTenant := config.Viper.GetString(config.Flag.Service.Azure.TenantID)
+
+		gsClientCredentialsConfig, err = credential.NewAzureCredentials(
+			config.Viper.GetString(config.Flag.Service.Azure.ClientID),
+			config.Viper.GetString(config.Flag.Service.Azure.ClientSecret),
+			mcTenant,
+		)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		// Need to set all Organization Credentials' tenant ID as aux tenant.
+		secretList := &v1.SecretList{}
+		err = k8sClient.CtrlClient().List(context.Background(), secretList, ctrl.MatchingLabels{label.App: "credentiald"})
```

Comment:
Don't these secrets ever get added, deleted or modified? This implementation runs forever with the state that existed when the program started. Shouldn't these be loaded on-demand when needed, in order to have fresh configuration of current state? Now if one removes credentials from MC, the running `azure-operator` still keeps using them.

Reply:
I believe this force us to restart the operator every time a Organization with a new tenant is created. I think it's not the best solution, but taking into account that I don't think this happens a lot, and that eventually we will stop running azure-operator, I'd say it's ok-ish?

Reply:
you are both right.
The current setup of azure operator creates one single azure client set with the MC credentials rather than a factory like it's done with the organization secrets.
It would be a huge refactoring to change how that works and I don't think it's worth the effort considering, as Jose said, that azure operator will eventually disappear.
I understand the need to restart azure operator when org secrets change to be an issue but:
- secret change is handled by SEs only
- this should be a pretty rare situation.

Are you ok with this compromise or do you want me to work on a proper solution?

Reply:
I'm ok with leaving this now as-is. Let's hope we don't hit issues with this. A workaround to make this reliable could be to add a controller watching these specific secrets and when it notices that known list of credentials changes, it would `os.Exit(1)` in order to restart the operator.

Reply:
yeah that was my thinking as well.
You know what? it's should be easy enough, I will give it a try

Reply:
Just a wish: Can you submit it in a separate PR so it's easier to grasp? :hugs: